src/scene/layer.js

```diff
@@ -659,6 +659,10 @@ Object.assign(pc, function () {
         var tempx, tempy, tempz;
         for (i = 0; i < drawCallsCount; i++) {
             drawCall = drawCalls[i];
+            if (drawCall.calculateSortDistance) {
```

Comment:
I believe this block should come after the two if statements below. Otherwise calculateSortDistance is modifying behavior other than zdist calc.

Reply:
I did consider where to put the block, and decided that I thought `calculateSortDistance` should be prioritized if supplied. However, your point that this modifies behavior other than zdist calc is valid.

I'll move the block down as you suggested and add to the documentation of `MeshInstance`, describing that the mesh must also be in one of some specific layers.

By reading the source code I was unable to fully grasp what `drawCall.command` really means. Could you enlighten me?

Reply:
Sure. It's a function callback where you can do anything - like maybe clear the depth buffer, for example.